subtests/docker_cli/attach/attach.py

```diff
@@ -36,10 +36,34 @@ def wait_interactive_cmd(self):
                      % wait)
         time.sleep(wait)
 
+    def image_exist(self, image_name):
+        di = DockerImages(self)
+        image_list = DockerImages.get_dockerimages_list(di)
+        exist_status = DockerImages.filter_list_full_name(image_list,
+                                                          image_name)
+        return exist_status
+
+    def pull_image(self, image_name):
+        print "pull_image"
+        dkrcmd = AsyncDockerCmd(self, 'pull', [image_name],
+                                self.config['docker_timeout'],
+                                verbose=True)
+        self.loginfo("Executing background command: %s" % dkrcmd)
+        dkrcmd.execute()
+        while not dkrcmd.done:
+            self.loginfo("Pulling...")
+            time.sleep(3)
```

Comment:
Just FYI: AsyncDockerCmd does not use ``timeout``, but it's okay in this case since test cannot continue w/o image and autotest will timeout the test if it takes too long.  One day I should fix this :confused: in ``AsyncDockerCmd``